sql/core/src/main/scala/org/apache/spark/sql/classic/DataFrameWriter.scala

```diff
@@ -484,12 +484,19 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) extends sql.DataFram
           serde = None,
           external = false,
           constraints = Seq.empty)
+        val writeOptions = lookupV2Provider() match {
```

Comment:
nit: `def saveAsTableCommand` already calls `lookupV2Provider()` and we can pass the value to here?